sql/walrus--0.1.sql

```diff
@@ -37,18 +36,22 @@ create type realtime.user_defined_filter as (
 );
 
 
+create function realtime.to_regrole(role_name text)
+    returns regrole
+    immutable
+    language sql
+    -- required to allow use in generated clause
+as $$ select role_name::regrole $$;
+
+
 create table realtime.subscription (
-    -- Tracks which users are subscribed to each table
-    id bigint not null generated always as identity,
-    user_id uuid not null,
-    -- Populated automatically by trigger. Required to enable auth.email()
-    email varchar(255),
+    -- Tracks which subscriptions are active
+    id uuid primary key,
     entity regclass not null,
     filters realtime.user_defined_filter[] not null default '{}',
-    created_at timestamp not null default timezone('utc', now()),
-
-    constraint pk_subscription primary key (id),
-    unique (entity, user_id, filters)
```

Comment:
@olirice

> -- Apply consistent order to filters so the unique constraint on
> -- (subscription_id, entity, filters) can't be tricked by a different filter order

should we add back a unique index here for (id, entity, filters)?

Reply:
The `id` alone is unique so it wouldn't help with duplicate subscriptions by a single user anymore.

We were previously able to do this because we had a reference to the `user_id`, that would be constant for multiple subscriptions

removed filter reordering here 71567088253a2c853a5adb3df4a1e774f28914fb

Reply:
@olirice ah, you're absolutely right!